### PR TITLE
Renames Anchors to Links

### PR DESCRIPTION
**Pull request by iMerica**
#### What does this PR do?
Addresses #3297 

#### Where should the reviewer start?
Run tests. Observe they all pass. Observe that Anchor is still importable and backwards compatible.

#### What testing has been done on this PR?
Unit testing. All tests pass.

#### How should this be manually tested?
Import grommet into a new project and try using Link like you normally would Anchor. Also use Anchor like normal and anything that is upstream from Anchor.

#### Do the grommet docs need to be updated?
Theres definitely still vestiges of Anchor in the docs. I'll leave this PR as WIP for now.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Yes.

### CODE REVIEW

**Comment by oorestisime** [comment]
Few thoughts:

This is not backwards compatible. I see an easy solution to fix this:

- Create a HOC of Link called Anchor that console warns during development that this component is deprecated and user should migrate to Link.

How about we provide a codemod for easy migration of users. Here's an example https://github.com/reactjs/react-codemod

While we are at it, how about we also provide the `to` prop that forwards to href. this should help solve some issues discussed in slack.

File: .eslintignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 node_modules/**
 dist/**
-coverage/**
\ No newline at end of file
+coverage/**
+*-test.js
+*stories.js
```
Comment (oorestisime) [inline .eslintignore]: em, why this change?

File: src/js/components/Anchor/index.js [hunk 2]
```diff
@@ -1 +0,0 @@
-export { Anchor } from './Anchor';
```
Comment (oorestisime) [inline src/js/components/Anchor/index.js]: this is where i propose to add the HOC

export const Anchor = (props) => {
  if dev -> warn

  return <Link {...props} />
}

File: src/js/components/Link/StyledLink.js [hunk 3]
```diff
@@ -24,30 +24,30 @@ const sizeStyle = props => {
   `;
 };
 
-const StyledAnchor = styled.a`
+const StyledLink = styled.a`
   box-sizing: border-box;
   ${props => sizeStyle(props)}
   color: ${props =>
-    normalizeColor(props.colorProp || props.theme.anchor.color, props.theme)};
+    normalizeColor(props.colorProp || props.theme.link.color, props.theme)};
```
Comment (oorestisime) [inline src/js/components/Link/StyledLink.js]: can we also make these changes backwards compatible? easiest way to do is first check in theme.link. if its not available we search in theme.anchor.
and lets add a comment in this file to remove this compatibility check in next major

File: src/js/components/Link/index.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export { Link } from './Link';
+export { Link as Anchor } from './Link';
```
Comment (oorestisime) [inline src/js/components/Link/index.js]: see proposal up for also warning about using this in development phase

File: src/js/components/Link/StyledLink.js [hunk 5]
```diff
@@ -24,30 +24,30 @@ const sizeStyle = props => {
...
   ${props =>
-    props.theme.anchor.fontWeight &&
-    `font-weight: ${props.theme.anchor.fontWeight};`}
+    props.theme.link.fontWeight &&
```
[the remark this answers is not in the document]
Comment (oorestisime) [reply src/js/components/Link/StyledLink.js]: yeah my idea also was to create a helper function to get to access theme level and when we give back anchor we warn.

**Comment by iMerica** [comment]
@ShimiSun Yep. I'm having a look now ...

**Comment by iMerica** [comment]
> I'm wondering whether we should add the deprecation warning message now. We have so many examples out there that will start showing the warning and it will take a while to update them. I'm wondering if we should not add the warning until we've allowed some time for the examples to be updated first?

Done.

**Comment by iMerica** [comment]
It looks like the only thing pending is style component related backwards compatibility. If someone who is savvy with styled components would like to handle this, I would appreciate it. 

You can add your commit to this PR.

File: src/js/components/Link/StyledLink.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (oorestisime) [reply src/js/components/Link/StyledLink.js]: small note, we should only warn in development

**Comment by oorestisime** [comment]
I am picking this then thanks @iMerica :)

**Comment by oorestisime** [comment]
Hey i can't seem to push in this PR. i created a new one https://github.com/grommet/grommet/pull/3873

IMHO we should link to the issue and have a clear resolution there (maybe lock discussion as well) so that users who see the warning know what to do. @ShimiSun ideas?

**Comment by ShimiSun** [comment]
I'll be closing this PR, we'll be tracking it on #3873
